src/storage.rs

```diff
@@ -1149,19 +1157,90 @@ pub(crate) fn data_dir_to_db_path(data_dir: &Path) -> PathBuf {
     data_dir.join("db")
 }
 
+/// Stores the compression setting to a metadata file.
+///
+/// # Errors
+///
+/// Returns an error if the file cannot be created or written to.
+fn store_compression_metadata(data_dir: &Path, compression: bool) -> Result<()> {
+    let metadata_path = data_dir.join("COMPRESSION");
```

Comment:
Please, add information about the COMPRESSION metadata file to the README.md
file.